arrow-data/src/test/kotlin/arrow/data/OptionTest.kt

```diff
@@ -137,8 +137,14 @@ class OptionTest : UnitSpec() {
 
         "exists" {
             some.exists { it.startsWith('k') } shouldBe true
+            some.exists { it.startsWith('j') } shouldBe false
             none.exists { it.startsWith('k') } shouldBe false
```

Comment:
I debated changing the predicate in both None cases to throw a runtime exception, to additionally assert that the predicate itself is never called. Is there already an existing mechanism for this type of assertion in arrow?

Reply:
@benfleis if you are referring to do that in the test you can use `kotlin-test` facilities https://github.com/kotlintest/kotlintest/blob/master/doc/reference.md#exceptions- to capture exceptions.

Reply:
the exception will propagate and fail the tests. If you'd like to fail the test manually use fail("Message").

Reply:
If I update none.exists as follows:

```kotlin
    none.exists { throw NotImplementedError("predicate must not run") } shouldBe false
```

it strengthens the test. I was curious whether that throw above (a la `TODO(...)`) is the standard. Thus, it should not throw -- this variant would simply make it apparent if None's implementation became wrong.

Reply:
I'd rather it was a simple `fail()` call as it's part of the framework. Won't block on it tho.

Reply:
ah, that's what I was looking for. Have to run, but will try to drop a tiny PR for this later.